src/game/AI/ScriptDevAI/scripts/world/scourge_invasion.cpp

```diff
@@ -269,13 +269,28 @@ uint32 GetFindersAmount(Creature* shard)
 /*
 Circle
 */
+
+std::map<uint32, std::set<ObjectGuid>> CirclePositions;
```

Comment:
Global mutable variables do not behave like you think they do. This particular variable will be thread unsafe, because both kalimdor and EK will try to alter it at the same time. Only permissible global variables are const ones. If you want to store it, you need to store it as part of world_map_kalimdor and world_map_eastern_kingdoms. The easiest way probably is to  void OnObjectCreate(GameObject*) {} add this and add it to m_goEntryGuidCollection, and then fetch it from there. It will not be sorted by zone but that can always be checked on use. You will no longer need to do it in GoCircle constructor and just fetch it when needed.